src/Filter/ClassFilter.php

```diff
@@ -45,33 +45,22 @@ public function filter(ProxyQueryInterface $query, string $alias, string $field,
     public function getDefaultOptions(): array
     {
         return [
+            'field_type' => ChoiceType::class,
             'operator_type' => EqualOperatorType::class,
             'operator_options' => [],
         ];
     }
 
-    public function getFieldType(): string
-    {
-        return $this->getOption('field_type', ChoiceType::class);
-    }
-
-    public function getFieldOptions(): array
-    {
-        $choiceOptions = [
-            'required' => false,
-            'choices' => $this->getOption('sub_classes'),
-        ];
-
-        return $this->getOption('choices', $choiceOptions);
-    }
-
     public function getRenderSettings(): array
     {
         return [DefaultType::class, [
             'operator_type' => $this->getOption('operator_type'),
             'operator_options' => $this->getOption('operator_options'),
             'field_type' => $this->getFieldType(),
-            'field_options' => $this->getFieldOptions(),
+            'field_options' => [
```

Comment:
I didn't kept the choices option.

I think it's not used. It's supposed to be used this way
```
'choices' => [
    'choices' => [...],
],
```
The syntax is so weird I assume it wasn't used this way.
But I wonder if I shouldn't add `choice_translation_domain => false` in the default values.